### PR TITLE
Inject list of beans

### PR DESCRIPTION
**Pull request by yuriyfomin17**
-  List-based injection
- Comments added
- Tests for List
- Some code refactoring
- Moved same logic from `injectBeanViaSetter` and `injectBeanViaConstructor` to `assignBeans`

### CODE REVIEW

File: src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java [hunk 1]
```diff
@@ -222,10 +280,31 @@ private <T> Predicate<Map.Entry<String, Object>> filterByBeanType(Class<T> beanT
...
+    /**
+     * <h3>Finds bean from Application Context</h3>
+     * <p>Special form of getBean method if we expect to receive multiple beans</p>
+     *
+     * @param beanType        {@link Class}
+     * @param oneBeanExpected {@link Boolean}
+     * @return {@link Object}
+     */
+
+    public <T> T getBean(Class<T> beanType, boolean oneBeanExpected) throws NoSuchBeanException, NoUniqueBeanException {
```
Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java]: I think second param not necessary here.

File: src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java [hunk 2]
```diff
@@ -159,45 +160,102 @@ private <T> void injectBeanViaAutowiredAnnotation(Class<?> beanClass, T beanInst
...
+     * <p>Assigns the bean instance to the parent Class field </p>
+     * <p>{@link AnnotationConfigApplicationContext#multipleBeansExpected(Class)} is used to check whether {@link List} of beans are expected</p>
+     * <p>If list is expected then {@link AnnotationConfigApplicationContext#getElementListType(Class)} is used to determine parameter of the elements in {@link List}</p>
+     * <p>If one bean expected, {@link Field} is assigned with bean instance value</p>
+     *
+     * @param beanClass {@link Class}
+     * @param beanInstance {@link Object}
+     * @param fieldTypeStringNameMap {@link Map}
+     */
+    private <T> void assignBeans(Class<?> beanClass, T beanInstance, Map<Class<?>, String> fieldTypeStringNameMap, Class<?> parameterType) throws NoSuchFieldException, IllegalAccessException {
```
Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java]: Better to rename to injectBeans. What do you think?

**Review by Mikhail-Mikhailenko — COMMENTED** [review]
Could you please rosolve conflits.

File: src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java (hunk 1 above)
Comment (yuriyfomin17) [reply src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java]: actually it is . pull the branch and doulbe check

File: src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java (hunk 2 above)
Comment (yuriyfomin17) [reply src/main/java/com/breskul/bring/AnnotationConfigApplicationContext.java]: agree. will rename

**Pull request merged**